toxav/msi.c

```diff
@@ -792,35 +868,61 @@ static void handle_pop(MSICall *call, const MSIMessage *msg)
 
     kill_call(call);
 }
-static void handle_msi_packet(Messenger *m, uint32_t friend_number, const uint8_t *data, uint16_t length, void *object)
+
+/* !!hack!! */
+MSISession *tox_av_msi_get(void *av);
+
+static void handle_msi_packet(Tox *tox, uint32_t friend_number, const uint8_t *data, size_t length2, void *object)
 {
-    LOGGER_DEBUG(m->log, "Got msi message");
+    if (length2 < 2) {
+        // we need more than the ID byte for MSI messages
+        return;
+    }
+
+    // Zoff: is this correct?
+    uint16_t length = (uint16_t)(length2 - 1);
+
+    // Zoff: do not show the first byte, its always "PACKET_ID_MSI"
+    const uint8_t *data_strip_id_byte = (const uint8_t *)(data + 1);
+
+    LOGGER_API_DEBUG(tox, "Got msi message");
+
+    void *toxav = tox_get_av_object(tox);
```

Comment:
Given that this is a hack, we can at least make it a bit more type-safe hack. Can you change all these `void`s to `ToxAV`? You'll need to forward-declare it in tox.c so we can have a `ToxAV*` there instead of `void*`. The hack also becomes a bit more obvious that way.

Reply:
not sure if i can do that. i tried and got into a circular problem